src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/InteropServices/UseValidPlatformStringTest.cs

```diff
@@ -0,0 +1,389 @@
+        If OperatingSystem.IsOSPlatform(""windows"") Then PlatformSpecificAPI()
+        If OperatingSystem.IsOSPlatformVersionAtLeast(""ios"", 12, 100, 0, 1) Then PlatformSpecificAPI()
+        If OperatingSystem.IsOSPlatform([|""windows8.0""|]) Then PlatformSpecificAPI()
+    End Sub
+
+    Public Sub PlatformSpecificAPI()
+    End Sub
+End Class";
+            await VerifyAnalyzerAsyncVb(vbSource);
+        }
+
+        [Fact]
+        public async Task MoreInvalidPlatformStringsWarns()
+        {
+            var source = @"
+using System.Runtime.Versioning;
+
+public class Test
+{
+    [[|SupportedOSPlatform(""Ios-4.1"")|]]
+    public void UnsupportedOSPlatformIosDash4_1() { }
+
+    [[|SupportedOSPlatform(""Ios*4.1"")|]]
+    public void UnsupportedOSPlatformIosStar4_1() { }
+
+    [[|SupportedOSPlatform(null)|]]
+    public void UnsupportedOSPlatformWithNullString() { }
+
+    [[|SupportedOSPlatform(""Linux_4.1"")|]]
+    public void SupportedLinux_41() { }
+
+    [[|UnsupportedOSPlatform("""")|]]
+    public void UnsupportedWithEmptyString() { }
+}";
+            await VerifyAnalyzerAsyncCs(source);
+        }
+
+        private static async Task VerifyAnalyzerAsyncCs(string sourceCode)
+        {
+            var test = PopulateTestCs(sourceCode);
+            await test.RunAsync();
+        }
+
+        private static async Task VerifyAnalyzerAsyncCs(string sourceCode, string editorconfigText)
+        {
+            var test = PopulateTestCs(sourceCode);
+            test.TestState.AdditionalFiles.Add((".editorconfig", editorconfigText));
+            await test.RunAsync();
+        }
+
+        private static VerifyCS.Test PopulateTestCs(string sourceCode, params DiagnosticResult[] expected)
+        {
+            var test = new VerifyCS.Test
+            {
+                TestCode = sourceCode,
+                ReferenceAssemblies = ReferenceAssemblies.Net.Net50,
+                MarkupOptions = MarkupOptions.UseFirstDescriptor,
+                TestState = { }
+            };
+            test.ExpectedDiagnostics.AddRange(expected);
+            return test;
+        }
+
+        private static async Task VerifyAnalyzerAsyncVb(string sourceCode)
+        {
+            var task = PopulateTestVb(sourceCode);
+            await task.RunAsync();
+        }
+
+        private static async Task VerifyAnalyzerAsyncVb(string sourceCode, string editorconfigText)
+        {
+            var test = PopulateTestVb(sourceCode);
+            test.TestState.AdditionalFiles.Add((".editorconfig", editorconfigText));
+            await test.RunAsync();
+        }
+
+        private static VerifyVB.Test PopulateTestVb(string sourceCode, params DiagnosticResult[] expected)
+        {
+            var test = new VerifyVB.Test
+            {
+                TestCode = sourceCode,
+                ReferenceAssemblies = ReferenceAssemblies.Net.Net50,
+                MarkupOptions = MarkupOptions.UseFirstDescriptor,
+                TestState = { },
```

Comment:
Is this needed? If so, maybe consider a comment?

Reply:
I think it was needed when setting `params DiagnosticResult[]`, but for these tests that is not used, i might remove it